src/it/scala/com/exasol/cloudetl/kafka/KafkaTopicDataImporterAvroToColumnsIT.scala

```diff
@@ -41,6 +46,54 @@ class KafkaTopicDataImporterAvroToColumnsIT extends KafkaTopicDataImporterAvroIT
     )
   }
 
+  test("run emits records when the starting offset of the topic is greater zero") {
+    createCustomTopic(topic)
+
+    val startingOffset = 12
+    0.until(startingOffset)
+      .foreach(recordNr => {
+        publishToKafka(
+          topic,
+          AvroRecord(
+            "Some record that we delete" +
+              "to ensure the offset does not start at zero",
+            recordNr,
+            13
+          )
+        )
+      })
+    deleteRecordsFromTopic(topic, startingOffset)
+
+    publishToKafka(topic, AvroRecord("abc", 3, 13))
+    publishToKafka(topic, AvroRecord("hello", 4, 14))
+
+    val iter = mockExasolIterator(properties, Seq(0), Seq(-1))
```

Comment:
I think this test would work without new changes, right?

I think `earliest` helps for this scenario, please correct me if I am wrong:

- import into Exasol records 1, 2, 3 (offsets)
- delete records 1, 2 from topic
- add new records to topic
- import should get new records with offsets 1, 2

Or is earliest is relevant when we import into empty Exasol table, e.g, the very first time.

Reply:
No, without the change of the default offset strategy to "earliest" (via com.exasol.cloudetl.kafka.KafkaConsumerProperties#AUTO_OFFSET_RESET) the test fails and this the bug I am trying to fix.

The problem ist that once you call seek on a consumer with an invalid offset, it applies the supplied `auto.offset.reset` strategy. You can check this when using the console consumer with log level info:

```
Fetch position FetchPosition{offset=1, offsetEpoch=Optional.empty, currentLeader=LeaderAndEpoch{leader=Optional[xxxx:9092 (id: 19322 rack: null)], epoch=21}} is out of range for partition xxxxx-1, resetting offset (org.apache.kafka.clients.consumer.internals.Fetcher)
```

Since the offset of a topic is an ever increasing number and retention is removing old messages, the start offset can be any number, so for
* Initial load into empty tables using offset 0 most times this is not valid and the consumer starts with strategy `latest` 
* Any load that starts with a "last offset" smaller than the current minimum offet of the topic, the same happens. This can happen when time- or byte-based retention is faster than the interval the import job runs

Does this make sense?

Reply:
Yes, it totally makes sense, thanks for the explanation!